src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -183,7 +189,9 @@ public IClient build() {
 			dispatcher.setMaxRequests(maxRequests);
 			dispatcher.setMaxRequestsPerHost(maxRequestsPerHost);
 
-			OkHttpClient.Builder builder = new OkHttpClient.Builder()
+			OkHttpClient.Builder builder = null;
+			if (!this.sslCertCallbackWithDefaultHostnameVerifier) {
```

Comment:
I'll try to clarify.  The point of the change is the `.hostnameVerifier(this.sslCertificateCallback)` call.  If we make this call, we bypass the use of the default hostname verifier, okhttp3.internal.tls.OkHostnameVerifier, and force implementors to duplicate all the good work okhttp3 did in supporting http://www.ietf.org/rfc/rfc2818.txt and https://tools.ietf.org/html/rfc6125 in the `ISSLCertificateCallback.allowHostname` call.

The jenkins-plugin is fine with letting okhttp3 handle hostname verification.  But we still want to leverage the `ISSLCertificateCallback.allowCertificate` callback when setting up the trust store in com.openshift.restclient.ClientBuilder.getCurrentTrustManager(TrustManagerFactory).